File: ProgrammingIdeas/Adapters/CategoryAdapter.cs
using Android.Content;
using Android.Support.V7.Widget;
using Android.Views;
using Android.Widget;
using System;
using System.Collections.Generic;

namespace ProgrammingIdeas.Adapters
{
	public class CategoryAdapter : RecyclerView.Adapter
	{
		private List<Category> categories;
		public Action<int> ItemClick;

		private int[] icons = {Resource.Mipmap.numbers, Resource.Mipmap.text, Resource.Mipmap.network, Resource.Mipmap.enterprise,
									   Resource.Mipmap.cpu, Resource.Mipmap.web, Resource.Mipmap.file, Resource.Mipmap.database,
									   Resource.Mipmap.multimedia, Resource.Mipmap.games};

		public CategoryAdapter(List<Category> category)
		{
			categories = category;
		}

		public override int ItemCount => categories.Count;

		public override void OnBindViewHolder(RecyclerView.ViewHolder holder, int position)
		{
			var category = categories[position];

			var viewHolder = holder as CategoryViewholder;
			viewHolder.CategoryLabel.Text = category.CategoryLbl;
			viewHolder.IdeasCount.Text = $"Ideas: {category.CategoryCount}";
			viewHolder.Icon.SetImageResource(icons[position]);
			viewHolder.Root.SetBackgroundColor(Android.Graphics.Color.Transparent);
			if (position == Global.CategoryScrollPosition)
				viewHolder.Root.SetBackgroundResource(Resource.Color.highlight);
		}

		public override RecyclerView.ViewHolder OnCreateViewHolder(ViewGroup parent, int viewType)
		{
			View row = LayoutInflater.From(parent.Context).Inflate(Resource.Layout.categoryrow, parent, false);
			return new CategoryViewholder(row, ItemClick);
		}
	}

	public class CategoryViewholder : RecyclerView.ViewHolder
	{
		public ImageView Icon { get; set; }
		public TextView CategoryLabel { get; set; }
		public TextView IdeasCount { get; set; }
		public RelativeLayout Root { get; set; }

		public CategoryViewholder(View itemView, Action<int> listener) : base(itemView)
		{
			itemView.Click += (sender, e) => listener?.Invoke(AdapterPosition);
			Icon = itemView.FindViewById<ImageView>(Resource.Id.categoryIcon);
			CategoryLabel = itemView.FindViewById<TextView>(Resource.Id.categoryLbl);
			IdeasCount = itemView.FindViewById<TextView>(Resource.Id.ideaCountLbl);
			Root = itemView.FindViewById<RelativeLayout>(Resource.Id.categoryRoot);
		}
	}
}

File: android/kotlin/app/src/main/java/com/alansa/ideabag2/adapters/CategoryAdapter.kt
package com.alansa.ideabag2.adapters

import android.graphics.Color
import android.support.v4.content.ContextCompat
import android.support.v7.widget.RecyclerView
import android.view.LayoutInflater
import android.view.ViewGroup
import android.widget.ImageView
import com.alansa.ideabag2.Global
import com.alansa.ideabag2.R
import com.alansa.ideabag2.databinding.RowCategoryBinding
import com.alansa.ideabag2.models.Category

class CategoryAdapter(private val categories: List<Category>, var itemClick: (Int) -> Unit) : RecyclerView.Adapter<CategoryViewHolder>() {
    private val icons = listOf(R.mipmap.numbers, R.mipmap.text, R.mipmap.network, R.mipmap.enterprise,
            R.mipmap.cpu, R.mipmap.web, R.mipmap.file, R.mipmap.database, R.mipmap.multimedia, R.mipmap.games)

    override fun getItemCount(): Int {
        return categories.size
    }

    override fun onBindViewHolder(holder: CategoryViewHolder, position: Int) {
        holder.bind(categories[position], itemClick, icons[position]);
    }

    override fun onCreateViewHolder(parent: ViewGroup, viewType: Int): CategoryViewHolder {
        var binding = RowCategoryBinding.inflate(LayoutInflater.from(parent.context), parent, false)
        return CategoryViewHolder(binding);
    }
}

class CategoryViewHolder(val binding: RowCategoryBinding) : RecyclerView.ViewHolder(binding.categoryRoot) {
    fun bind(category: Category, itemClick: (Int) -> Unit, categoryIcon: Int) {
        binding.category = category;
        binding.categoryRoot.setBackgroundColor(Color.TRANSPARENT)
        binding.categoryRoot.setOnClickListener { itemClick(adapterPosition) }
        binding.categoryRoot.findViewById<ImageView>(R.id.categoryIcon).setImageResource(categoryIcon)

        if (Global.categoryClickIndex == adapterPosition)
            binding.categoryRoot.setBackgroundColor(ContextCompat.getColor(binding.categoryRoot.context, R.color.highlight))
    }
}